pandas/tests/test_expressions.py

```diff
@@ -95,7 +95,10 @@ def run_binary(self, df, other):
                 expr.get_test_result()
                 result = op(df, other)
                 used_numexpr = expr.get_test_result()
-                assert used_numexpr, "Did not use numexpr as expected."
+
+                # We don't currently use numexpr for comparisons in Series,
```

Comment:
hmm, the original and entire purpose IS to use numexpr for comparisions. I agree consistency is important, but this is turned off for DataFrames? (for regular dtypes)????

Reply:
yah, the right place to do that is in array_ops (which will cause is to do it for Series too)

I've got a branch that implements dataframe-with-dataframe ops block-wise, but the diff for that is pretty big, so im splitting pieces off.  the numexpr-for-series-comparisons part will be next

Reply:
ok great

yeah comparison ops should. r a pretty big speedup as it’s parallle
computation

Reply:
ok, im in the process of splitting off the relevant parts of that branch.  That may end up affecting the tests here, but the core.ops edits here will be correct regardless.

I think we need to improve the used_numexpr check in the tests to not just check whether numexpr was called, but whether it actually was used effectively

Reply:
also xref #31297 will smooth this out